packages/cli/lib/projects.js

```diff
@@ -522,9 +522,27 @@ const handleProjectUpload = async (
 
   archive.pipe(output);
 
-  archive.directory(srcDir, false, file =>
-    shouldIgnoreFile(file.name, true) ? false : file
-  );
+  let loggedIgnoredNodeModule = false;
+
+  archive.directory(srcDir, false, file => {
+    const ignored = shouldIgnoreFile(file.name, true);
+    if (ignored) {
+      const isNodeModule = file.name.includes('node_modules');
+
+      if (!isNodeModule || !loggedIgnoredNodeModule) {
+        logger.debug(
+          i18n(`${i18nKey}.handleProjectUpload.fileFiltered`, {
+            filename: file.name,
+          })
+        );
+      }
+
+      if (isNodeModule && !loggedIgnoredNodeModule) {
+        loggedIgnoredNodeModule = true;
+      }
+    }
+    return ignored ? false : file;
```

Comment:
This isn't a blocker, but I'm thinking about the TypeScript headache here when we eventually convert the CLI to TS.

Reply:
This is a good call out, but I think I may skip changing it for now. I took a look at the [docs](https://www.archiverjs.com/docs/archiver/#directory) and couldn't find any obvious ways to make this better without potentially introducing some obscure bug.